core/src/main/java/org/elasticsearch/index/query/BoolQueryBuilder.java

```diff
@@ -207,4 +323,96 @@ private void doXArrayContent(String field, List<QueryBuilder> clauses, XContentB
+
+        booleanQuery.setBoost(this.boost);
+        Queries.applyMinimumShouldMatch(booleanQuery, this.minimumShouldMatch);
+        Query query = this.adjustPureNegative ? fixNegativeQueryIfNeeded(booleanQuery) : booleanQuery;
+        if (this.queryName != null) {
+            parseContext.addNamedQuery(this.queryName, query);
+        }
+        return query;
+    }
+
+    @Override
+    public QueryValidationException validate() {
+        // nothing to validate, clauses are optional, see hasClauses(), other parameters have defaults
+        return null;
+    }
+
+    private void addBooleanClauses(QueryParseContext parseContext, BooleanQuery booleanQuery, List<QueryBuilder> clauses, Occur occurs)
+            throws IOException {
+        for (QueryBuilder query : clauses) {
+            Query luceneQuery = query.toQuery(parseContext);
+            if (luceneQuery != null) {
+                booleanQuery.add(new BooleanClause(luceneQuery, occurs));
+            }
+        }
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(boost, adjustPureNegative, disableCoord,
+                minimumShouldMatch, queryName, mustClauses, shouldClauses, mustNotClauses, filterClauses);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        BoolQueryBuilder other = (BoolQueryBuilder) obj;
+        return Objects.equals(boost, other.boost) &&
+                Objects.equals(adjustPureNegative, other.adjustPureNegative) &&
+                Objects.equals(disableCoord, other.disableCoord) &&
+                Objects.equals(minimumShouldMatch, other.minimumShouldMatch) &&
+                Objects.equals(queryName, other.queryName) &&
+                Objects.equals(mustClauses, other.mustClauses) &&
+                Objects.equals(shouldClauses, other.shouldClauses) &&
+                Objects.equals(mustNotClauses, other.mustNotClauses) &&
+                Objects.equals(filterClauses, other.filterClauses);
+    }
+
+    @Override
+    public BoolQueryBuilder readFrom(StreamInput in) throws IOException {
+        BoolQueryBuilder boolQueryBuilder = new BoolQueryBuilder();
+        boolQueryBuilder.must(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.mustNot(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.should(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.filter(QueryBuilder.readQueryBuilderListFrom(in));
+        boolQueryBuilder.boost = in.readFloat();
+        boolQueryBuilder.queryName = in.readOptionalString();
+        boolQueryBuilder.adjustPureNegative = in.readBoolean();
+        boolQueryBuilder.disableCoord = in.readBoolean();
+        boolQueryBuilder.minimumShouldMatch = in.readOptionalString();
+        return boolQueryBuilder;
```

Comment:
In that case (e.g. for boost), why do we currently use primitives instead of Objects (e.g. Float). In that case it would be clear if a value was provided by the user or we should use the default. Just curious what the reasons for using primitives in many places like this are.